tests/integration/golang/mlflow/namespace/flows/experiment_test.go

```diff
@@ -0,0 +1,598 @@
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace2Code,
+		).WithQuery(
+			request.GetExperimentRequest{
+				ID: experiment1ID,
+			},
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ExperimentsRoutePrefix, mlflow.ExperimentsGetRoute),
+		),
+	)
+	assert.Equal(
+		s.T(),
+		fmt.Sprintf(
+			"RESOURCE_DOES_NOT_EXIST: unable to find experiment '%s': error getting experiment by id: %s: record not found",
+			experiment1ID,
+			experiment1ID,
+		),
+		resp.Error(),
+	)
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+
+	resp = api.ErrorResponse{}
+	assert.Nil(
+		s.T(),
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithQuery(
+			request.GetExperimentRequest{
+				ID: experiment2ID,
+			},
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ExperimentsRoutePrefix, mlflow.ExperimentsGetRoute),
+		),
+	)
+	assert.Equal(
+		s.T(),
+		fmt.Sprintf(
+			"RESOURCE_DOES_NOT_EXIST: unable to find experiment '%s': error getting experiment by id: %s: record not found",
+			experiment2ID,
+			experiment2ID,
+		),
+		resp.Error(),
+	)
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+
+	// 4. test `GET /experiments/get-by-name` endpoint.
+	// check that experiments were created in scope of difference namespaces.
+	s.getExperimentByNameCompare(
+		namespace1Code,
+		experiment1.Experiment.Name,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:               experiment1ID,
+				Name:             "ExperimentName1",
+				Tags:             []response.ExperimentTagPartialResponse{},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+	s.getExperimentByNameCompare(
+		namespace2Code,
+		experiment2.Experiment.Name,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:               experiment2ID,
+				Name:             "ExperimentName2",
+				Tags:             []response.ExperimentTagPartialResponse{},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+
+	// 5. test `GET /experiments/search` endpoint.
+	s.searchExperimentAndCompare(namespace1Code, []*response.ExperimentPartialResponse{
+		experiment1.Experiment,
+	})
+	s.searchExperimentAndCompare(namespace2Code, []*response.ExperimentPartialResponse{
+		experiment2.Experiment,
+	})
+
+	// 6. test `POST /experiments/update` endpoint.
+	s.updateExperiment(namespace1Code, &request.UpdateExperimentRequest{
+		ID:   experiment1.Experiment.ID,
+		Name: "UpdatedExperiment1",
+	})
+	s.updateExperiment(namespace2Code, &request.UpdateExperimentRequest{
+		ID:   experiment2.Experiment.ID,
+		Name: "UpdatedExperiment2",
+	})
+
+	// check that experiments were updated.
+	s.getExperimentByIDAndCompare(
```

Comment:
There is so much repetition of the method parameters in these tests, can they be wrapped into something simpler to read?

Reply:
they are different and they use to compare `expected` and `actual` results so I don't know what he could be simplified.